src/components/application_manager/src/plugin_manager/rpc_plugin_manager_impl.cc

```diff
@@ -44,17 +61,27 @@ RPCPluginPtr LoadPlugin(const std::string& full_plugin_path) {
   }
 
   typedef RPCPlugin* (*Create)();
-  Create create_plugin = reinterpret_cast<Create>(dlsym(plugin_dll, "Create"));
-  char* error_string = dlerror();
-  if (nullptr != error_string) {
-    LOG4CXX_ERROR(logger_,
-                  "Failed to export dll's " << full_plugin_path
-                                            << " symbols : " << error_string);
+  Create create_plugin = GetFuncFromLib<Create>(plugin_dll, "Create");
+  if (!create_plugin) {
+    LOG4CXX_ERROR(logger_, "No Create function in " << full_plugin_path);
+    return std::move(empty_plugin);
+  }
+
+  typedef void (*Delete)(RPCPlugin*);
+  Delete delete_plugin = GetFuncFromLib<Delete>(plugin_dll, "Delete");
```

Comment:
It seems all `Delete(RPCPlugin* plugin)` functions just do `delete plugin;`.
Do you foresee more logic being required in these `Delete` functions in the future?
I wonder if we can replace `delete_plugin(plugin);` in `plugin_destroyer` with just `delete plugin;` and save some exports.

Reply:
I can see 2 reasons why the exported Delete function is needed.

    The plugin may need an additional place to free allocated resources.
    There is a recommendation to delete memory from the same library that allocated this memory to avoid a potential crash.